File: PredictionAgent-Demo/src/nodes/base_node.py
"""
节点基类
"""

from abc import ABC, abstractmethod
from typing import Any, Dict


class BaseNode(ABC):
    """节点基类"""

    def __init__(self, llm_client, node_name: str = ""):
        self.llm_client = llm_client
        self.node_name = node_name or self.__class__.__name__

    @abstractmethod
    def run(self, input_data: Any, **kwargs) -> Any:
        """执行节点处理逻辑"""
        pass

    def log_info(self, message: str):
        """记录信息日志"""
        print(f"[{self.node_name}] {message}")

    def log_error(self, message: str):
        """记录错误日志"""
        print(f"[{self.node_name}] 错误: {message}")

File: PredictionAgent-Go/nodes/base_node.go
package nodes

import "fmt"

// BaseNode 节点基类
type BaseNode struct {
	LLMClient any
	NodeName  string
}

// NewBaseNode 创建节点
func NewBaseNode(llmClient any, nodeName string) *BaseNode {
	if nodeName == "" {
		nodeName = "BaseNode"
	}
	return &BaseNode{
		LLMClient: llmClient,
		NodeName:  nodeName,
	}
}

// LogInfo 记录信息日志
func (n *BaseNode) LogInfo(message string) {
	fmt.Printf("[%s] %s\n", n.NodeName, message)
}

// LogError 记录错误日志
func (n *BaseNode) LogError(message string) {
	fmt.Printf("[%s] Error: %s\n", n.NodeName, message)
}
